core/runtime/src/main/java/io/quarkus/runtime/annotations/RegisterForProxy.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.runtime.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that can be used to force an interface (including its super interfaces) to be registered for dynamic proxy
+ * generation in native image mode.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Repeatable(RegisterForProxy.List.class)
```

Comment:
Why should it be repeatable if `targets` already takes an array of values?

Reply:
@gastaldi AFAIK, when we register for proxy we add the interface and its super interfaces. For example:

```java
interface Foo extends Bar {}

...

Class<?>[] interfaces = new Class<?>[]{Foo.class, Bar.class};
Proxy.newProxyInstance(classLoader, interfaces, handler);
```

Even in `proxy-config.json`, each record is an array:

```json
[
 { "interfaces": [ "a.b.c.Foo", "a.b.c.Bar" ] },
 { "interfaces": [ "a.b.c.Baz" ] }
]
```

That's why the annotation is repeatable:

```java
@RegisterForProxy(targets = {Foo.class, Bar.class})
@RegisterForProxy(targets = Baz.class)
```